key_management/src/sigs.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::errors::Error;
+use address::Address;
+use bls_signatures::{PrivateKey as BlsPrivate, Serialize};
+use crypto::{Signature, SignatureType};
+use rand::rngs::OsRng;
+use secp256k1::{Message as SecpMessage, PublicKey as SecpPublic, SecretKey as SecpPrivate};
+
+/// Return the public key for a given private_key and SignatureType
+pub fn to_public(sig_type: SignatureType, private_key: Vec<u8>) -> Result<Vec<u8>, Error> {
+    match sig_type {
+        SignatureType::BLS => Ok(BlsPrivate::from_bytes(&private_key)
+            .map_err(|err| Error::Other(err.to_string()))?
+            .public_key()
+            .as_bytes()),
+        SignatureType::Secp256 => {
+            let private_key = SecpPrivate::parse_slice(private_key.as_ref())
+                .map_err(|err| Error::Other(err.to_string()))?;
+            let public_key = SecpPublic::from_secret_key(&private_key);
+            Ok(public_key.serialize().to_vec())
+        }
+    }
+}
+
+/// Return a new Address that is of a given SignatureType and uses the supplied public_key
+pub fn new_address(sig_type: SignatureType, public_key: Vec<u8>) -> Result<Address, Error> {
+    match sig_type {
+        SignatureType::BLS => {
+            let addr = Address::new_bls(public_key.as_ref())
+                .map_err(|err| Error::Other(err.to_string()))?;
+            Ok(addr)
+        }
+        SignatureType::Secp256 => {
+            let addr = Address::new_secp256k1(public_key.as_ref())
+                .map_err(|err| Error::Other(err.to_string()))?;
+            Ok(addr)
+        }
+    }
+}
+
+/// Sign takes in SignatureType, private key and message. Returns a Signature for that message
+pub fn sign(
+    sig_type: SignatureType,
+    private_key: Vec<u8>,
+    msg: Vec<u8>,
```

Comment:
This should be a slice, should always only need to read bytes to generate a signature